doc/design/new-user-questionnaire.md

```diff
@@ -0,0 +1,56 @@
+# Premise
+Currently, anyone with a GitHub account can signup for DANDI and make full use of its functionality. The goal of this design document is to introduce an
+approval process to help prevent misuse of the archive.
+
+## 1. Implementation Details
+### Changes to User model
+Three new columns, `status`, `questionnaire_form`, and `rejection_reason` will be introduced to the `User` model in dandi-api.
+- `status` is an enum that can be one of four values: `INCOMPLETE`, `PENDING`, `APPROVED`, or `REJECTED`.
+- `questionnaire_form` is a JSON field. The keys are questions, and the values are the users' answers.
+- `rejection_reason` is an optional string that contains a message written by one of the admins describing why a user was denied approval (if applicable).
+
+### Changes to API
+- /users/me needs to include the user's current status
+- A [Django group](https://docs.djangoproject.com/en/3.2/topics/auth/default/#groups), `approved_users`, will be created. Only users in this group will have access to protected endpoints.
+- Only approved users can create and edit dandisets.
```

Comment:
I assume that we would tweak the search endpoint to only return verified users. Alternatively, we would identify how to redesign the user search endpoints to provide different data streams for different purposes, including one that only returns verified users. (This could be as simple as filtering the full search endpoint with a query argument of some kind.)

Reply:
> I assume that we would tweak the search endpoint to only return verified users.

I think this would be sufficient for the needs of the web app and hub. The only place where the user search endpoint is used in the web app is for adding owners to a dandiset, where we don't want non-`APPROVED` users showing up either.

Reply:
should be verbalized ("only approved users returned"?) in text of the design doc.

Reply:
> should be verbalized ("only approved users returned"?) in text of the design doc.

https://github.com/dandi/dandi-api/commit/c8cde8bda07f8a069b98551068584ff311ade839